subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -1056,7 +1056,7 @@ static int tls_set_psk(struct tls_context *tls,
 		       struct tls_credential *psk,
 		       struct tls_credential *psk_id)
 {
-#if defined(MBEDTLS_KEY_EXCHANGE_SOME_PSK_ENABLED)
+#if defined(CONFIG_MBEDTLS_KEY_EXCHANGE_SOME_PSK_ENABLED)
```

Comment:
Yes, I think so, indeed I'm moving this to a separate commit so it's easier to backport it.

Reply:
Is this really a bug though? The symbols is defined by mbed TLS.

Reply:
Yes, `MBEDTLS_KEY_EXCHANGE_SOME_PSK_ENABLED` would be for TLS 1.2, but if we want to take also TLS 1.3 into account then we should change it to `MBEDTLS_SSL_HANDSHAKE_WITH_PSK_ENABLED`.
Now the question is: Zephyr has a similar kconfig `CONFIG_MBEDTLS_KEY_EXCHANGE_SOME_PSK_ENABLED` which seems to mimic the official Mbed TLS one and which I extended in this PR to take TLS 1.3 PSK key exchanges into account. I'm not a huge fan of duplicating things because it requires to keep the Zephyr kconfig version updated to all the changes that might happen to the Mbed TLS one. So can I remove the kconfig and rely only on the Mbed TLS symbol?

Reply:
I don't think we can remove that config, it serves as a dependency for `config MBEDTLS_PSK_MAX_LEN` at the Kconfig level.

> Yes, MBEDTLS_KEY_EXCHANGE_SOME_PSK_ENABLED would be for TLS 1.2

> Edit: in case we remove the kconfig, this is no more of interest for the backport to 3.7 of course

Sure, but for Zephyr 3.7 there's no TLS 1.3 support, so I'm still not convinced it's really needed to backport the change. Of course I might miss something, so if there's really a reason we could backport it, no big deal, just note that backporting changes require a corresponding bug report issue open.

Reply:
Checking more on this I found that `MBEDTLS_PSK_MAX_LEN` is actually a user configurable symbol in Mbed TLS (defined in `mbedtls_config.h`) which has a default of 32, whereas `MBEDTLS_KEY_EXCHANGE_SOME_PSK_ENABLED` and `MBEDTLS_SSL_HANDSHAKE_WITH_PSK_ENABLED` are both "internal" symbols, meaning that they _cannot_ be set from the user intentionally. This means that also in Mbed TLS it can happen that a user defines `MBEDTLS_PSK_MAX_LEN` to some non default value, but he does not set any PSK key exchange. This would be the same as if I remove the `CONFIG_MBEDTLS_KEY_EXCHANGE_SOME_PSK_ENABLED` in Zephyr and the dependency of `CONFIG_MBEDTLS_PSK_MAX_LEN` from it. This basically is what @tomi-font proposed here:

> remove it and either do nothing for MBEDTLS_PSK_MAX_LEN (would be okay and simplest to me)